src/core/net/dns_headers.cpp

```diff
@@ -373,5 +490,50 @@ otError Name::LabelIterator::ReadLabel(char *aLabelBuffer, uint8_t &aLabelLength
     return error;
 }
 
+bool Name::LabelIterator::CompareLabel(const char *&aName, bool aIsSingleLabel) const
+{
+    // This method compares the current label in the iterator with the
+    // `aName` string. `aIsSingleLable` indicates whether `aName` is a
+    // single label, or a sequence of labels separated by dot '.' char.
+    // If the label matches `aName`, then `aName` pointer is moved
+    // forward to the start of the next label (skipping over the `.`
+    // char). This method returns `true` when the labels match, `false`
+    // otherwise.
+
+    bool matches = false;
+
+    VerifyOrExit(StringLength(aName, mLabelLength) == mLabelLength);
```

Comment:
I think this is correct. @simonlingoogle can you please help double-check my reasons below (these are always tricky and good to have multiple eyes on them).

`aName` can be a single label or a dot separated name like `"label1.label2.label3"`. In the second case `aName` length can indeed be longer than `mLabelLength`.

- This statement is there to  check that the `aName` length is longer than `mLabelLength`
  - so that we can safely use `CompareBytes()` to read and compre `mLabelLength` bytes from `aName` (and avoid reading beyond the end of the string).
- Or saying it another way, this statement ensures that in the first `mLabelLength` chars of the `aName` array, there is no null char. 
  - `StringFind(aString, aMaxLength)` checks the first `aMaxLength` chars in the string char array against null char (e.g.,`StringLengrth(aString, 1)` would check only one index at `aString[0]`).
  -  So `StringFind(aName, mLabelLength)` checks the first `mLabelLength` chars in `mName`.

Reply:
Indeed it's correct. Thanks.